### PR TITLE
Removes wrong usage of atomic package

### LINKED ISSUE #109: We are using atomic.AddInt64 wrong
https://github.com/giantswarm/inago/blob/master/controller/update.go#L248

### PR DESCRIPTION
**Pull request by zeisss**
Fixes #109

### CODE REVIEW

**Comment by zeisss** [comment]
ping @giantswarm/cena

**Comment by JosephSalisbury** [comment]
LGTM

File: controller/update.go [hunk 1]
```diff
@@ -268,15 +266,13 @@ func (c controller) UpdateWithStrategy(ctx context.Context, req Request, opts Up
...
-					removeInProgress = v
+					atomic.AddInt64(&removeInProgress, 1)
 					err := c.removeFirst(ctx, newReq, opts)
 					if err != nil {
 						fail <- maskAny(err)
 						return
 					}
-					v = atomic.AddInt64(&removeInProgress, -1)
-					removeInProgress = v
+					atomic.AddInt64(&removeInProgress, -1)
```
Comment (xh3b4sd) [inline controller/update.go:275]: What was actually wrong here? I did it this weird way because go vet or something complained about this.

Comment (zeisss) [inline controller/update.go:275]: by assigning the variable again you have concurrent writes again and the whole usage of `atomic` is useless.

Comment (xh3b4sd) [inline controller/update.go:275]: Makes sense. Anyway I am really confused about this case and I looked that up what to do about it.

Comment (zeisss) [inline controller/update.go:275]: Not sure what you mean - do you want more explanation?

Comment (xh3b4sd) [inline controller/update.go:275]: I wan to remember the case I had to comprehend what happened and why it now is like I originally wanted to have it. Nothing to do for you here. Thanks.

**Pull request merged**